On The Map/OTMClient.swift

```diff
@@ -0,0 +1,192 @@
+                print(error)
+                completionHandlerForPOST(result: nil, error: error)
+            } else {
+                var newData = data
+                if(udacity){
+                    newData = data!.subdataWithRange(NSMakeRange(5, data!.length - 5)) 
+                }
+                /* 5/6. Parse the data and use the data (happens in completion handler) */
+                self.convertDataWithCompletionHandler(newData!, completionHandlerForConvertData: completionHandlerForPOST)
+            }
+            
+        }
+        
+        /* 7. Start the request */
+        task.resume()
+        
+        return task
+    }
+    
+    // MARK: - GET
+    
+    func taskForGETMethod(method: String, udacity: Bool, parameters: [String:AnyObject]?, completionHandlerForGET: (result: AnyObject!, error: NSError?) -> Void) -> NSURLSessionDataTask {
+        
+        var urlString: String
+        
+        if let mutableParameters = parameters {
+            urlString = method + OTMClient.escapedParameters(mutableParameters)
+        } else {
+            urlString = method
+        }
+        
+        let request = NSMutableURLRequest(URL: NSURL(string: urlString)!)
+        
+        if (!udacity) {
+            request.addValue("QrX47CA9cyuGewLdsL7o5Eb8iug6Em8ye0dnAbIr", forHTTPHeaderField: "X-Parse-Application-Id")
+            request.addValue("QuWThTdiRmTux3YaDseUSEpUKo7aBYM737yKd4gY", forHTTPHeaderField: "X-Parse-REST-API-Key")
+
+        }
+        
+        let task = session.dataTaskWithRequest(request) { (data, response, error) in
+            if let dataError = error {
+                OTMClient.errorForData(data, response: response, error: dataError)
+                completionHandlerForGET(result: nil, error: dataError)
+            } else {
+                var newData = data
+                if(udacity){
+                    newData = data!.subdataWithRange(NSMakeRange(5, data!.length - 5)) /* subset response data! */
+                }
+                self.convertDataWithCompletionHandler(newData!, completionHandlerForConvertData: completionHandlerForGET)
+            }
+        }
+        task.resume()
+        
+        return task
+    }
+    
+    // given raw JSON, return a usable Foundation object
+    private func convertDataWithCompletionHandler(data: NSData, completionHandlerForConvertData: (result: AnyObject!, error: NSError?) -> Void) {
+        
+        var parsedResult: AnyObject!
+        do {
+            parsedResult = try NSJSONSerialization.JSONObjectWithData(data, options: .AllowFragments)
+        } catch {
+            let userInfo = [NSLocalizedDescriptionKey : "Could not parse the data as JSON: '\(data)'"]
+            completionHandlerForConvertData(result: nil, error: NSError(domain: "convertDataWithCompletionHandler", code: 1, userInfo: userInfo))
+        }
+        
+        completionHandlerForConvertData(result: parsedResult, error: nil)
```

Comment:
If you don't use the `guard` statement suggestion above, this should be moved inside the `do` statement. Otherwise, when you have an error, the `completionHandlerforConverData` will be called twice.